ci-automation/vendor-testing/aws.sh

```diff
@@ -0,0 +1,121 @@
+MORE_INSTANCE_TYPES_VAR="AWS_${arch}_MORE_INSTANCE_TYPES"
+MORE_INSTANCE_TYPES=( ${!MORE_INSTANCE_TYPES_VAR} )
+
+if [[ -z "${AWS_OFFER}" ]]; then
+    AWS_OFFER='basic'
+fi
+
+case "${AWS_OFFER}" in
+    basic)
+        AWS_OEM_SUFFIX=''
+        ;;
+    pro)
+        AWS_OEM_SUFFIX='_pro'
+        ;;
+    *)
+        echo "1..1" > "${tapfile}"
+        echo "not ok - all AWS tests" >> "${tapfile}"
+        echo "  ---" >> "${tapfile}"
+        echo "  ERROR: Invalid offer '${AWS_OFFER}', should be 'basic' or 'pro'." | tee -a "${tapfile}"
+        echo "  ..." >> "${tapfile}"
+        exit 1
+        ;;
+esac
+
+testscript="$(basename "$0")"
+
+if [[ "${AWS_AMI_ID}" == "" ]]; then
+    echo "++++ ${testscript}: downloading flatcar_production_ami_vmdk${AWS_OEM_SUFFIX}_image.vmdk.bz2 for ${vernum} (${arch}) ++++"
+    copy_from_buildcache "images/${arch}/${vernum}/flatcar_production_ami_vmdk${AWS_OEM_SUFFIX}_image.vmdk.bz2" .
+
+    lbunzip2 "${work_dir}/flatcar_production_ami_vmdk${AWS_OEM_SUFFIX}_image.vmdk.bz2"
+
+    AWS_BUCKET="flatcar-kola-ami-import-${AWS_REGION}"
+    trap 'ore -d aws delete --region="${AWS_REGION}" --name="${AWS_IMAGE_NAME}" --ami-name="${AWS_IMAGE_NAME}" --file="${work_dir}/flatcar_production_ami_vmdk${AWS_OEM_SUFFIX}_image.vmdk" --bucket "s3://${AWS_BUCKET}/${AWS_BOARD}/"' EXIT
+    ore aws initialize --region="${AWS_REGION}" --bucket "${AWS_BUCKET}"
+    AWS_AMI_ID=$(ore aws upload --force --region="${AWS_REGION}" --name="${AWS_IMAGE_NAME}" --ami-name="${AWS_IMAGE_NAME}" --ami-description="Flatcar Test ${AWS_IMAGE_NAME}" --file="${work_dir}/flatcar_production_ami_vmdk${AWS_OEM_SUFFIX}_image.vmdk" --bucket  "s3://${AWS_BUCKET}/${AWS_BOARD}/" | jq -r .HVM)
+    echo "Created new AMI ${AWS_AMI_ID} (will be removed after testing)"
+fi
+
+run_aws_kola_test() {
+    local instance_type="${1}"
+    local instance_tapfile="${2}"
+
+    sudo \
+        --preserve-env=AWS_ACCESS_KEY_ID,AWS_SECRET_ACCESS_KEY \
+        timeout --signal=SIGQUIT 6h \
+        kola run \
+         --board="${AWS_BOARD}" \
+         --basename="${AWS_IMAGE_NAME}" \
+         --channel="${AWS_CHANNEL}" \
+         --offering="${AWS_OFFER}" \
+         --parallel="${AWS_PARALLEL}" \
+         --platform=aws \
+         --aws-ami="${AWS_AMI_ID}" \
+         --aws-region="${AWS_REGION}" \
+         --aws-type="${instance_type}" \
+         --aws-iam-profile="${AWS_IAM_PROFILE}" \
+         --tapfile="${instance_tapfile}" \
+         --torcx-manifest=../torcx_manifest.json \
+         "${@}"
+}
+
+cl_internet_included="$(kola list --platform=aws --filter "${@}" | { grep cl.internet || : ; } )"
+run_more_tests=0
+if [[ -n "${cl_internet_included}" ]] && [[ "${#MORE_INSTANCE_TYPES[@]}" -gt 0 ]]; then
+    run_more_tests=1
+fi
+if [[ "${run_more_tests}" -eq 1 ]]; then
+    for instance_type in "${MORE_INSTANCE_TYPES[@]}"; do
+        (
+            OUTPUT=$(run_aws_kola_test "${instance_type}" "validate_${instance_type}.tap" 'cl.internet' 2>&1 || :)
```

Comment:
Would be good to check how the sqlite logic handles the different `cl.internet` tests and doesn't hide the output of a failed cl.internet case for one instance type just because another instance type happened to have success

Reply:
We could postprocess the tap file to rename the cl.internet test to something like `$INSTANCETYPE.cl.internet`

Reply:
@pothos @krnowak what about something like that:
```bash
    timeout --signal=SIGQUIT 6h \
        kola run \
         --board="${AWS_BOARD}" \
         --basename="${AWS_IMAGE_NAME}" \
         --channel="${AWS_CHANNEL}" \
         --offering="${AWS_OFFER}" \
         --parallel="${AWS_PARALLEL}" \
         ...

    # compare the tested instance with the default instance type 
    if [[ "${instance_type}" != "${AWS_INSTANCE_TYPE}" ]]; then
        sed --in-place "s/cl\.internet/${instance_type}\.cl\.internet" "${instance_tapfile}"
    fi
```

Reply:
Yes, and some logic to run the test again for the right instance type if the rerun script passes `something.cl.internet` to rerun the failed test

Reply:
Seems this code would be good to go into https://github.com/flatcar-linux/scripts/pull/293

Reply:
I based this PR on #293, and added some functions for handling tap files and prefixed tests for restarting.